### PR TITLE
Feature: hooks refactor

### PR DESCRIPTION
**Pull request by D4nielJ**
## Hello reviewer :gem: 

This PR modifies: 
* Every class-based component to work as a functional component.

Adds: 
* Necessary hooks in components.
* New functionality to the display, now it shows current operator and holder value.

### Thank you for your feedback, is appreciated and keep being awesome! :rocket: 

P.S.: [Visit the live preview here](https://d4nielj.github.io/math-magicians/)

### CODE REVIEW

File: src/components/calc.js [hunk 1]
```diff
@@ -1,60 +1,42 @@
...
-    this.handleClick = this.handleClick.bind(this);
-    this.handleKeyDown = this.handleKeyDown.bind(this);
-  }
+const Calc = () => {
+  const [calc, setCalc] = useState({ total: null, next: null, operation: null });
 
-  handleClick = (e) => {
-    this.updateState(e.target.dataset.name);
+  const updateState = async (obj, key) => {
+    let { total, next, operation } = await calculate(obj, key);
```
Comment (Davidosky007) [inline src/components/calc.js]: * I get errors in the console when i perform division by zero operation (ie 8 / 0) kindly fix this error 🙏 
image

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
## Hi @D4nielJ 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/components/calc.js (hunk 1 above)
Comment (D4nielJ) [reply src/components/calc.js]: You got me! 😆 I'll fix this and I'll try to implement a max length for inputs aswell so numbers don't flow out of the box.

**Review by arslanbisharat — APPROVED** [review]
# Approved 👏 

Hi there, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of code reviews per project which you can see from your dashboard. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)_

**Pull request merged**